### ISSUE TITLE
Nicofox doesn't "see" videos and display download button anymore

### ISSUE THREAD

**Issue by Dioxaz** [littlebtc/nicofox#67]
Each time I think I found a workaround, this damn add-on breaks again.

Most of you would suggest me to let it go by now. But let me try again, probably for the last time.

My last successful download with nicofox happened Sunday. Then I wanted to download a video today, and for some unknown reason, nicofox is incapable to sense the presence of a video, even though I'm on the right tab and the video is loaded. No matter if I choose HTML5 or Flash, the result is the same. I'm granted with the "Welcome to Nicofox" message instead of a download button. I also tried with a fresh Firefox Portable and same result. I swear it used to work Sunday, as I downloaded a video.

If I'm insisting, this is because this add-on has no equivalent in terms of features. I liked having thumbs and comments for each video, and browse through them with it's own "Download manager". Indeed I can still download with DTA in a separate Firefox 52 but I repeat this add-on has no equivalent in terms of functionality today. So I think it's sad if it becomes useless.

edit: It looks like it can't sense that I'm logged in. I get the "Welcome to Nicofox" whether I'm logged in or not. Curiously, I can still resume older aborted downloads where it can actually "see" that I'm logged in. Another add-on, for Kodi, called XBMCnico yield similar behaviour now and keeps insisting that my email and password are incorrect (they are correct!).

**Comment by Teadog** [comment]
The reason for this seem to be that Nicovideo switched to HTTPS few days ago and NicoFox does not see such links as videos.  If you have a link in HTTP format then you can still download the video with "Download Link with NicoFox" context menu entry, but that's about it.

For example:
https://www.nicovideo.jp/watch/sm33913281 (Does not work)
http://www.nicovideo.jp/watch/sm33913281 (Works)

Nicofox needs to be modified to recognize HTTPS, though I'm not sure how to do this.

**Comment by Dioxaz** [comment]
That was a good call again, thank you very much. And I should have guessed it since the slow transition to HTTPS is breaking more and more stuff.

I was able to make the add-on work again by replacing "http" with "https" in all instances I could find which looked like a niconico URL. 

I've attached my modified add-on: [nicofox@littlebtc.zip](https://github.com/littlebtc/nicofox/files/2497186/nicofox.littlebtc.zip)

Now I have to invistigate and do the same on XBMCnico. It would be great if it was that simple (edit: not necessary, as its author already updated his add-on and it's now working again, at least for non-premium accounts).
